core/src/test/scala/org/locationtech/rasterframes/RasterJoinSpec.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * This software is licensed under the Apache 2 license, quoted below.
+ *
+ * Copyright 2019 Astraea, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     [http://www.apache.org/licenses/LICENSE-2.0]
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.locationtech.rasterframes
+
+import geotrellis.raster.io.geotiff.GeoTiff
+import geotrellis.raster.resample.Bilinear
+import geotrellis.raster.testkit.RasterMatchers
+import geotrellis.raster.{IntConstantNoDataCellType, Raster, Tile}
+import org.apache.spark.sql.functions._
+import org.locationtech.rasterframes.encoders.serialized_literal
+import org.locationtech.rasterframes.expressions.aggregates.TileRasterizerAggregate
+import org.locationtech.rasterframes.expressions.aggregates.TileRasterizerAggregate.ProjectedRasterDefinition
+
+
+class RasterJoinSpec extends TestEnvironment with TestData with RasterMatchers {
+  import spark.implicits._
+  describe("Raster join between two DataFrames") {
+    val s1 = readSingleband("L8-B4-Elkton-VA.tiff")
+    val s2 = readSingleband("L8-B4-Elkton-VA-4326.tiff")
```

Comment:
This tiff has no nodata defined on it,  and in debugger I see cellType is `uint16raw` but there are nodata artifacts from the different griddings. See the black areas with values of 0 in the attached. 

The minimum value in the original file is about 6000.

![image](https://user-images.githubusercontent.com/7798319/58493272-8f163c80-8140-11e9-9e46-7cc91d6452e3.png)

Reply:
I think the 0 / nodata issue here does not actually affect any of the tests currently written.

Reply:
Still, I wish L8 had a NoData value set it in it... PITA.